src/main/scala/viper/gobra/frontend/info/implementation/property/Visibility.scala

```diff
@@ -0,0 +1,205 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyr (c) 2011-2020 ETH Zurich.
+
+package viper.gobra.frontend.info.implementation.property
+
+import viper.gobra.ast.frontend._
+import viper.gobra.frontend.info.base.{SymbolTable => symb}
+import viper.gobra.frontend.info.implementation.TypeInfoImpl
+import viper.gobra.util.Violation
+
+trait Visibility extends BaseProperty { this: TypeInfoImpl =>
+
+  def isPrivate(expr: PExpression): Boolean = isPrivateExp(expr)
+
+  private def isPrivateExpTyp(etyp: PExpressionOrType): Boolean = etyp match {
+    case exp: PExpression => isPrivateExp(exp)
+    case typ: PType => isPrivateTyp(typ)
+    case t => Violation.violation(s"Unexpected Expression or Type in isPrivateExpTyp, got $t") 
+  }
+
+  def isPrivateReg(id: PIdnNode): Boolean = regular(id) match {
+    case g: symb.GlobalVariable => isPrivateString(g.id.name)
+    case _: symb.Variable => false 
```

Comment:
Variables are also inside specifications and therefore, can also refer to the arguments inside the function. Since the arguments of a function are always visible, they should be always public in this case.

Reply:
I do not understand this. Can you give me an example that is rejected if local variables were private?

Reply:
For example:
```
requires a == 0
func Foo(a int)
```
Here `a int` is still an instance of `symb.Variable`. So here if I would match `symb.Variable => true` it would throw an error expected public ... error. The same applies if Foo has returns and receiver arguments. 

I have a look at the code and several classes extend `symb.Variable`. I rewrote the single case of `symb.Variable` to:
```
case _: symb.InParameter => false
case _: symb.OutParameter => false
case _: symb.ReceiverParameter => false
case v: symb.Variable => v match {
  case v: symb.SingleLocalVariable => v.exp.exists(isPrivateExp)
  case v: symb.MultiLocalVariable => isPrivateExp(v.exp)
  case _ => false
}
```
I do not know if there are more relevant cases, since I have not encountered them yet.